swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/work/WorkProcessor.kt

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2018 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.swabbie.work
+
+import com.netflix.spectator.api.Registry
+import com.netflix.spinnaker.kork.lock.LockManager
+import com.netflix.spinnaker.swabbie.LockingService
+import com.netflix.spinnaker.swabbie.ResourceTypeHandler
+import com.netflix.spinnaker.swabbie.events.Action
+import com.netflix.spinnaker.swabbie.model.WorkItem
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression
+import org.springframework.scheduling.annotation.Scheduled
+import org.springframework.stereotype.Component
+import java.time.Clock
+
+@Component
+@ConditionalOnExpression("\${swabbie.enabled:true}")
+class WorkProcessor(
+  val clock: Clock,
+  val registry: Registry,
+  private val resourceTypeHandlers: List<ResourceTypeHandler<*>>,
+  private val workQueue: WorkQueue,
+  private val lockingService: LockingService
+) {
+  private val log: Logger = LoggerFactory.getLogger(javaClass)
+  private val workId = registry.createId("swabbie.work")
+
+  /**
+   * Takes work [WorkItem] off the [WorkQueue], acquires a lock and
+   * dispatches the corresponding work configuration to a suitable handler for processing
+   */
+  @Scheduled(fixedDelayString = "\${swabbie.work.intervalMs:5000}")
```

Comment:
`5000Ms` too tight for a default. Will bump that up

Reply:
Ya I agree, checking once every minute seems good?

Reply:
A min a two. Definitely not 5sec for a default.